src/sdk/namespace_fs.js

```diff
@@ -2040,6 +2038,82 @@ class NamespaceFS {
     // INTERNALS //
     ///////////////
 
+    /**
+     * _get_restore_status is a pure function which consumes the xattr
+     * of the target file and returns its current state.
+     * @param {nb.NativeFSXattr} xattr xattr of the target file
+     * @param {Date} now
+     * @returns {{ request?: number; expiry?: Date; state: nb.RestoreState } | null}
+     */
+    _get_restore_status(xattr, now) {
+        // Total 6 states (2x restore_request, 3x restore_expiry)
+        let restore_request;
+        let restore_expiry;
+
+        if (xattr[GlacierBackend.XATTR_RESTORE_REQUEST]) {
+            const request = parseInt(xattr[GlacierBackend.XATTR_RESTORE_REQUEST], 10);
+            if (request === Number(xattr[GlacierBackend.XATTR_RESTORE_REQUEST])) {
+                restore_request = request;
+            }
+        }
+        if (xattr[GlacierBackend.XATTR_RESTORE_EXPIRY]) {
+            const expiry = new Date(xattr[GlacierBackend.XATTR_RESTORE_EXPIRY]);
+            // @ts-ignore - Checks for `Invalid Date Object`
+            if (!isNaN(expiry)) {
```

Comment:
I think linter would prefer this check -
```suggestion
            if (!isNaN(expiry.getTime())) {
```